Tests/BrowserServicesKitTests/SmarterEncryption/HTTPSUpgradeReferenceTests.swift

```diff
@@ -0,0 +1,137 @@
+//
+//  HTTPSUpgradeReferenceTests.swift
+//  DuckDuckGo
+//
+//  Copyright © 2022 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import XCTest
+import os.log
+@testable import BrowserServicesKit
+@testable import BloomFilterWrapper
+
+private struct HTTPSUpgradesRefTests: Decodable {
+    struct HTTPSUpgradesTests: Decodable {
+        let name: String
+        let desc: String
+        let tests: [HTTPSUpgradesTest]
+    }
+    
+    struct HTTPSUpgradesTest: Decodable {
+        let name: String
+        let siteURL: String
+        let requestURL: String
+        let requestType: String
+        let expectURL: String
+        let exceptPlatforms: [String]
+        
+        var shouldSkip: Bool { exceptPlatforms.contains("ios-browser") }
+    }
+    
+    let navigations: HTTPSUpgradesTests
+    let subrequests: HTTPSUpgradesTests
+}
+
+// swiftlint:disable force_try
+// swiftlint:disable force_cast
+final class HTTPSUpgradeReferenceTests: XCTestCase {
```

Comment:
Can we also have tests to ensure user unprotected, temp unprotected and feature exceptions work?